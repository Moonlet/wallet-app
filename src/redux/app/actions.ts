import { HintsComponent, HintsScreen } from './state';
import { IReduxState } from '../state';
import { clearPinCode } from '../../core/secure/keychain';

export const SET_ACCEPTED_TC_VERSION = 'SET_ACCEPTED_TC_VERSION';
export const SHOW_HINT = 'SHOW_HINT';
export const RESET_FAILED_LOGINS = 'RESET_FAILED_LOGINS';
export const INCREMENT_FAILED_LOGINS = 'INCREMENT_FAILED_LOGINS';
export const SET_APP_BLOCK_UNTIL = 'SET_APP_BLOCK_UNTIL';
export const RESET_ALL_DATA = 'RESET_ALL_DATA';

export const appSetAcceptedTcVersion = (version: number) => {
    return {
        type: SET_ACCEPTED_TC_VERSION,
        data: version
    };
};

export const updateDisplayedHint = (screen: HintsScreen, component: HintsComponent) => {
    return {
        type: SHOW_HINT,
        data: { screen, component }
    };
};

export const incrementFailedLogins = () => {
    return {
        type: INCREMENT_FAILED_LOGINS
    };
};

export const resetFailedLogins = () => {
    return {
        type: RESET_FAILED_LOGINS
    };
};

export const setAppBlockUntil = (date: Date | string) => {
    return {
        type: SET_APP_BLOCK_UNTIL,
        data: { date }
    };
};

export const resetAllData = () => async (dispatch, getState: () => IReduxState) => {
    clearPinCode(); // clear keychain storage
    dispatch({ type: RESET_ALL_DATA });
};
