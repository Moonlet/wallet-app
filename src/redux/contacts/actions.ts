import { IContactState } from './state';

// actions consts
export const CONTACT_ADD = 'CONTACT_ADD';

export const addContact = (contactData: IContactState) => {
    return {
        type: CONTACT_ADD,
        data: contactData
    };
};
