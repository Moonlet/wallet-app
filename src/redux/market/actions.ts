export const EXCHANGE_RATE_UPDATE = 'EXCHANGE_RATE_UPDATE';

export const updateExchangeRate = (exchangeRate: { token: string; value: string }) => {
    return {
        type: EXCHANGE_RATE_UPDATE,
        data: exchangeRate
    };
};
