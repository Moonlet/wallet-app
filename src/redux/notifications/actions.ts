// actions consts
export const SET_HAS_UNSEEN_NOTIFICATIONS = 'SET_HAS_UNSEEN_NOTIFICATIONS';
export const SET_NOTIFICATIONS = 'SET_NOTIFICATIONS';
export const MARK_SEEN = 'MARK_SEEN';

export const setHasUnseenNotifications = (hasUnseenNotifications: boolean) => {
    return {
        type: SET_HAS_UNSEEN_NOTIFICATIONS,
        data: { hasUnseenNotifications }
    };
};

export const setNotifications = (notifications: any) => {
    return {
        type: SET_NOTIFICATIONS,
        data: { notifications }
    };
};

export const markSeenNotification = (blockchain: string, notifId: string) => {
    return {
        type: MARK_SEEN,
        data: { blockchain, notifId }
    };
};
