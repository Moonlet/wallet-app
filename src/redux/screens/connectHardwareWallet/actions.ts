// actions consts
export const VERIFY_ADDRESS_ON_DEVICE = 'VERIFY_ADDRESS_ON_DEVICE';
export const FEATURE_NOT_SUPPORTED = 'FEATURE_NOT_SUPPORTED';
export const TO_INITIAL_STATE = 'TO_INITIAL_STATE';

// actions creators
export const verifyAddressOnDevice = (verify: boolean) => {
    return {
        type: VERIFY_ADDRESS_ON_DEVICE,
        data: verify
    };
};

export const featureNotSupported = () => {
    return {
        type: FEATURE_NOT_SUPPORTED
    };
};

export const toInitialState = () => {
    return {
        type: TO_INITIAL_STATE
    };
};
