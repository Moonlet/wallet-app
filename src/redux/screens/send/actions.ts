// actions consts
export const REVIEW_TRANSACTION = 'REVIEW_TRANSACTION';

// actions creators
export const reviewTransaction = () => {
    return {
        type: REVIEW_TRANSACTION
    };
};
