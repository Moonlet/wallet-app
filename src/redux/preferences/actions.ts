import { Dispatch } from 'react';
import { IReduxState } from '../state';
import { getSelectedAccount, getSelectedBlockchain } from '../wallets/selectors';
import { getBalance, setSelectedBlockchain } from '../wallets/actions';
import { Blockchain, ChainIdType } from '../../core/blockchain/types';
import { IBlockchainsOptions } from './state';
import { IAction } from '../types';
import { getBlockchains, hasNetwork } from './selectors';

// actions consts
export const PREF_SET_CURRENCY = 'PREF_SET_CURRENCY';
export const TOGGLE_TOUCH_ID = 'TOGGLE_TOUCH_ID';
export const SET_TEST_NET = 'SET_TEST_NET';
export const PREF_SET_BLOCKCHAIN_ACTIVE_STATE = 'PREF_SET_BLOCKCHAIN_ACTIVE_STATE';
export const PREF_SET_BLOCKCHAIN_ORDER = 'PREF_SET_BLOCKCHAIN_ORDER';
export const PREF_SET_NETWORK_TEST_NET_CHAIN_ID = 'PREF_SET_NETWORK_TEST_NET_CHAIN_ID';

export const setBlockchainActive = (blockchain: Blockchain, active: boolean) => (
    dispatch: Dispatch<IAction>,
    getState: () => IReduxState
) => {
    const state = getState();
    const selectedBlockchain = getSelectedBlockchain(state);

    if (selectedBlockchain === blockchain && active === false) {
        const nextBlockchain = Object.keys(state.preferences.blockchains).find(
            object => object !== blockchain
        );
        if (nextBlockchain) {
            setSelectedBlockchain(Blockchain[nextBlockchain])(dispatch, getState);
        }
    }
    dispatch({
        type: PREF_SET_BLOCKCHAIN_ACTIVE_STATE,
        data: { blockchain, active }
    });
};

export const setBlockchainOrder = (blockchains: IBlockchainsOptions[]) => {
    return {
        type: PREF_SET_BLOCKCHAIN_ORDER,
        data: { blockchains }
    };
};

export const setNetworkTestNetChainId = (blockchain: Blockchain, chainId: ChainIdType) => {
    return {
        type: PREF_SET_NETWORK_TEST_NET_CHAIN_ID,
        data: { blockchain, chainId }
    };
};

export const toggleTestNet = () => (dispatch: Dispatch<any>, getState: () => IReduxState) => {
    dispatch({
        type: SET_TEST_NET
    });
    const state = getState();

    const selectedBlockchain = getSelectedBlockchain(state);
    const networkExists = hasNetwork(Blockchain[selectedBlockchain], state.preferences.testNet);

    if (!networkExists) {
        const blockchains = getBlockchains(state);
        setSelectedBlockchain(Blockchain[blockchains[0]])(dispatch, getState);
    }

    const selectedAccount = getSelectedAccount(getState());
    getBalance(
        selectedAccount.blockchain,
        selectedAccount.address,
        undefined,
        true
    )(dispatch, getState);
};

export function toggleTouchID() {
    return {
        type: TOGGLE_TOUCH_ID
    };
}

export function setCurrency(currency: string) {
    return {
        type: PREF_SET_CURRENCY,
        data: { currency }
    };
}
