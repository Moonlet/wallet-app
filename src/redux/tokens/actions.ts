import { IAccountState } from '../wallets/state';
import { ITokenConfigState } from './state';
import { Dispatch } from 'react';
import { IReduxState } from '../state';
import { addTokenToAccount } from '../wallets/actions';
import { generateAccountTokenState } from './static-selectors';
import { getChainId } from '../preferences/selectors';
import { Blockchain, ChainIdType } from '../../core/blockchain/types';
import { ApiClient } from '../../core/utils/api-client/api-client';
import { flattenObject } from '../utils/helpers';
import {
    addBreadcrumb as SentryAddBreadcrumb,
    captureException as SentryCaptureException
} from '@sentry/react-native';

export const ADD_TOKEN = 'ADD_TOKEN';
export const UPDATE_TOKEN_CONTRACT_ADDRESS = 'UPDATE_TOKEN_CONTRACT_ADDRESS';
export const REMOVE_TOKEN = 'REMOVE_TOKEN';

export const addToken = (account: IAccountState, token: ITokenConfigState) => (
    dispatch: Dispatch<any>,
    getState: () => IReduxState
) => {
    const blockchain = account.blockchain;
    const chainId = getChainId(getState(), blockchain).toString();
    addTokenForBlockchain(blockchain, token, chainId)(dispatch, getState);
    addTokenToAccount(account, generateAccountTokenState(token, account))(dispatch, getState);
};

export const addTokenForBlockchain = (
    blockchain: Blockchain,
    token: ITokenConfigState,
    chainId: ChainIdType
) => (dispatch: Dispatch<any>, getState: () => IReduxState) => {
    dispatch({
        type: ADD_TOKEN,
        data: { token, chainId, blockchain }
    });
};

export const updateTokenContracts = () => async (
    dispatch: Dispatch<any>,
    getState: () => IReduxState
) => {
    const state = getState();

    if (state.tokens) {
        const flatObject = flattenObject(state.tokens, 2);
        const keys = [];

        Object.keys(flatObject).map(key => {
            const strings = key.split('.');
            strings.splice(strings.length - 1, 0, 'tokens');
            strings.splice(0, 1, strings[0].toLowerCase());
            keys.push(strings.join('.'));
        });

        try {
            const configs = await new ApiClient().configs.getConfigs(keys);

            if (configs.result) {
                Object.keys(configs.result).map(key => {
                    const values = key.split('.');
                    const chainId = values[1];
                    const blockchain = values[0].toUpperCase();
                    const tokenSymbol = values[3];
                    const contractAddress = configs.result[key];
                    dispatch({
                        type: UPDATE_TOKEN_CONTRACT_ADDRESS,
                        data: { chainId, blockchain, tokenSymbol, contractAddress }
                    });
                });
            }
        } catch (error) {
            SentryAddBreadcrumb({
                message: JSON.stringify({
                    error
                })
            });

            SentryCaptureException(
                new Error(`Failed to update token contracts, ${error?.message}`)
            );
        }
    }
};
