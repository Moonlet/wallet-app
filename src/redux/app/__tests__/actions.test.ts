import { appSetTosVersion } from '../actions';
import { Blockchain } from '../../../core/blockchain/types';

describe('app actions', () => {
    test('app action creators', () => {
        expect(appSetTosVersion(1)).toMatchSnapshot();
    });
});
