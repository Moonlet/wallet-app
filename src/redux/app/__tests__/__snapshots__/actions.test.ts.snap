// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app actions app action creators 1`] = `
Object {
  "data": "1",
  "type": "APP_STATE_SWITCH_WALLET",
}
`;

exports[`app actions app action creators 2`] = `
Object {
  "data": 1,
  "type": "APP_SET_TOS_VERSION",
}
`;

exports[`app actions app action creators 3`] = `
Object {
  "data": Object {
    "blockchain": "ETHEREUM",
    "index": 0,
  },
  "type": "APP_SWITCH_ACCOUNT",
}
`;
